sql/signal_handler.cc

```diff
@@ -58,49 +58,70 @@ extern const char *optimizer_switch_names[];
 
 static inline void output_core_info()
 {
-  /* proc is optional on some BSDs so it can't hurt to look */
-#if defined(HAVE_READLINK) && !defined(__APPLE__) && !defined(__FreeBSD__)
+#ifndef __WIN__
   char buff[PATH_MAX];
   ssize_t len;
   int fd;
+  if ((fd= open("/etc/os-release", O_RDONLY)) >= 0)
+  {
+    len= read(fd, buff, sizeof(buff));
+    close(fd);
+    /* Take first 3 lines of this file, enough for distro and version only */
+    char *endline= (char *) memchr(buff, '\n', len);
+    len-= (endline - buff);
+    if (len > 0)
+    {
+      endline= (char *) memchr(endline + 1, '\n', len);
+      len-= (endline - buff);
+    }
+    if (len > 0)
+    {
+      endline= (char *) memchr(endline + 1, '\n', len);
+      len-= (endline - buff);
+    }
+    *endline= '\0';
+    my_safe_printf_stderr("\nDistribution information:\n%.*s\n\n", (int) len, buff);
+  }
+#endif
+  /* proc is optional on some BSDs so it can't hurt to look */
+#if defined(HAVE_READLINK) && !defined(__APPLE__) && !defined(__FreeBSD__)
   if ((len= readlink("/proc/self/cwd", buff, sizeof(buff))) >= 0)
   {
     my_safe_printf_stderr("Writing a core file...\nWorking directory at %.*s\n",
                           (int) len, buff);
   }
-  if ((fd= open("/proc/self/limits", O_RDONLY, MYF(0))) >= 0)
+  if ((fd= open("/proc/self/limits", O_RDONLY)) >= 0)
   {
     my_safe_printf_stderr("Resource Limits:\n");
-    while ((len= read(fd, (uchar*)buff, sizeof(buff))) > 0)
+    while ((len= read(fd, buff, sizeof(buff))) > 0)
     {
       my_write_stderr(buff, len);
     }
     close(fd);
```

Comment:
For each close I would add check if successful.

Reply:
On `man 2 close` "See NOTES for a discussion of why close() should not be retried after an error".  So there's no meaningful action to take if unsuccessful.